samples/highcharts/studies/pointandfigure/demo.js

```diff
@@ -0,0 +1,250 @@
+                lastPoint = getLastPoint(finalData);
+
+            if (upTrend) {
+
+                if (close - lastPoint >= boxSize) { // Add point going UP
+                    pushNewPoint(close, upTrend, lastPoint);
+                }
+
+                if (lastPoint - close >= reversal) { // Handle reversal
+                    upTrend = false;
+
+                    pushPointGroup(x, [], false);
+                    pushNewPoint(close, upTrend, lastPoint);
+                }
+            }
+
+            if (!upTrend) {
+
+                if (lastPoint - close >= boxSize) { // Add point going DOWN
+                    pushNewPoint(close, upTrend, lastPoint);
+                }
+
+                if (close - lastPoint >= reversal) { // Handle reversal
+                    upTrend = true;
+
+                    pushPointGroup(x, [], true);
+                    pushNewPoint(close, upTrend, lastPoint);
+                }
+            }
+        });
+    }
+
+    // Process the finalData to groupData return format
+    const groupedXData = [],
+        groupedYData = [],
+        groupMap = [];
+
+    finalData.forEach(point => {
+        const x = point.x;
+
+        point.y.forEach(y => {
+            groupedXData.push(x);
+            groupedYData.push(y);
+            groupMap.push({
+                start: x,
+                length: point.y.length,
+                options: {
+                    marker: point.symbol
+                }
+            });
+
+        });
+    });
+
+
+    return {
+        groupedXData,
+        groupedYData,
+        groupMap
+    };
+}
+
+Highcharts.wrap(Highcharts.Axis.prototype, 'getClosest', function (proceed) {
+    let ret = proceed.apply(this, Array.prototype.slice.call(arguments, 1));
+
+    const pnfSeries = this.series.filter(series => series.is('pointandfigure'));
+
+    pnfSeries.forEach(() => {
+        if (this.categories) {
+            ret = 1;
+        } else {
+            this.series.forEach(function (series) {
+                const seriesClosest = series.closestPointRange,
+                    visible = series.visible ||
+                        !series.chart.options.chart.ignoreHiddenSeries;
+
+                if (Highcharts.defined(seriesClosest) && visible) {
+                    ret = Highcharts.defined(ret) ?
+                        Math.min(ret, seriesClosest) :
+                        seriesClosest;
+                }
+            });
+        }
+    });
+    return ret;
+});
+// eslint-disable-next-line no-underscore-dangle
+Highcharts.wrap(Highcharts._modules['Core/Series/Series.js'].prototype, 'groupData', function (proceed) {
+    if (this.is('pointandfigure')) {
+        return generatePnfData.apply(
+            this, Array.prototype.slice.call(arguments, 1
+            ));
+    }
+    return proceed.apply(this, Array.prototype.slice.call(arguments, 1));
+
+});
+
+Highcharts.seriesType(
+    'pointandfigure',
+    'scatter', {
+    // Options
```

Comment:
Missing default options for:
- `boxSize` (I think we agreed on percentage, like 5%?)
- `reversalAmount` (defaults to 3x, right?)